main.cpp

```diff
@@ -152,6 +155,10 @@ static int parse_argv(int argc, char *argv[], const char **uart, char **udp_addr
             }
             break;
         }
+        case 'l': {
+            opt.logs_dir = strdup(optarg);
```

Comment:
Where do you free it? Anyway, as `optarg` lives while program lives, if you are not changing it, no need to `strdup` it.